src/credentials/FabricTable.h

```diff
@@ -347,9 +347,11 @@ class DLL_EXPORT FabricTable
         virtual ~Delegate() {}
 
         /**
-         * Gets called when a fabric is deleted, such as on FabricTable::Delete().
+         * Gets called when a fabric is changed in a significant way, such as cert being updated or fabric being delete.
+         * `fabricDeleted` indicates if the fabric has been deleted in case delegate treats updates differently from
+         * updates.
          **/
-        virtual void OnFabricDeletedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+        virtual void OnFabricHasChanged(FabricTable & fabricTable, FabricIndex fabricIndex, bool fabricDeleted) = 0;
```

Comment:
Fabric changes result from one of two operations:
- `RemoveFabric`, which will always effectively delete the fabric at a specific fabric index.
- `UpdateNOC`, which is effectively a 'speculative delete' of the accessing fabric's index, followed by the allocation of a new fabric index containing the new fabric.

If so, from the perspective of this call, `fabricIndex` is effectively _always_ getting deleted no, since the fabric at the existing fabric-index is no longer there/present?

Reply:
I stand corrected. @tcarmelveilleux's latest PR re-uses the same fabric-index when processing `UpdateNOC`.

Reply:
`UpdateNOC, which is effectively a 'speculative delete' of the accessing fabric's index, followed by the allocation of a new fabric index containing the new fabric.`

UpdateNOC was always "with the same fabric" and no idea of deletion